opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -373,15 +377,25 @@ private ExprValue parseStruct(Content content, String prefix, boolean supportArr
     content
         .map()
         .forEachRemaining(
-            entry ->
+            entry -> {
+              String fieldKey = entry.getKey();
+              String fullFieldPath = makeField(prefix, fieldKey);
+              try {
                 populateValueRecursive(
                     result,
-                    new JsonPath(entry.getKey()),
-                    parse(
-                        entry.getValue(),
-                        makeField(prefix, entry.getKey()),
-                        type(makeField(prefix, entry.getKey())),
-                        supportArrays)));
+                    new JsonPath(fieldKey),
+                    parse(entry.getValue(), fullFieldPath, type(fullFieldPath), supportArrays));
+              } catch (IllegalArgumentException e) {
+                // Return null for invalid field names (e.g., fields consisting only of dots)
+                // This can happen with disabled object fields that bypass field name validation
+                // Log warning to hint users about corrupt data
+                LOG.warn(
+                    "Field '{}' has invalid name and will return null: {}",
```

Comment:
This is going to be a lot of logging if the user ingests a lot of data like this. Maybe not too bad after compression though?